### PR TITLE
add 2 second delay to button on delete confirm page

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this adds a little delay so you don't accidentally click the delete confirm button

### CODE REVIEW

File: shared/settings/delete-confirm/index.desktop.js [hunk 1]
```diff
@@ -1,33 +1,58 @@
...
+    this.state = {allowDeleteForever: false}
+  }
+
+  componentWillMount () {
+    this.setState({allowDeleteForever: false})
+  }
+
+  componentDidMount () {
+    setTimeout(() => {
+      this.setState({allowDeleteForever: true})
```
Comment (chromakode) [inline shared/settings/delete-confirm/index.desktop.js]: Will this throw if the component is destroyed before the `setTimeout` fires? Alex created https://github.com/keybase/client/blob/master/shared/common-adapters/hoc-timers.js to ensure that `clearTimeout` is called when a component is unmounted, might be a use case here?

Comment (chrisnojima) [inline shared/settings/delete-confirm/index.desktop.js]: good catch, totally forgot about that. updated

**Comment by chrisnojima** [comment]
@chromakode want to look at the hoc update

**Comment by chrisnojima** [comment]
yeah, i think there's some good ideas where we can expand our usage. this is a good read: https://medium.com/@franleplant/react-higher-order-components-in-depth-cf9032ee6c3e#.2mz82mdri

**Pull request merged**